game/scripts/vscripts/Sections/Chapter4/SectionWards.ts

```diff
@@ -50,65 +65,105 @@ function onStart(complete: () => void) {
+                    hero.SetAttackCapability(UnitAttackCapability.NO_ATTACK);
                     hero.AddNewModifier(undefined, undefined, "modifier_invisible", undefined);
+                    hero.FaceTowards(playerHero.GetAbsOrigin());
+                    setUnitPacifist(hero, true);
                 }
             }),
 
-            tg.wait(1),
-
-            // Spawn wards and wait for player to pick them up. Also highlight wards during this.
             tg.withHighlights(tg.seq([
                 tg.immediate(_ => {
                     CreateItemOnPositionSync(wardLocationObs, observerWardItem);
                     CreateItemOnPositionSync(wardLocationSentry, sentryWardItem);
                 }),
 
                 tg.immediate(_ => goalFetchWard.start()),
+                tg.audioDialog(LocalizationKey.Script_4_Wards_1, LocalizationKey.Script_4_Wards_1, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
 
                 tg.completeOnCheck(_ => playerHero.HasItemInInventory("item_ward_dispenser"), 1),
             ]), { type: "arrow", locations: [wardLocationObs, wardLocationSentry] }),
 
+            tg.immediate(_ => goalFetchWard.complete()),
+            tg.audioDialog(LocalizationKey.Script_4_Wards_2, LocalizationKey.Script_4_Wards_2, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
+            tg.audioDialog(LocalizationKey.Script_4_Wards_3, LocalizationKey.Script_4_Wards_3, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
+            tg.audioDialog(LocalizationKey.Script_4_Wards_4, LocalizationKey.Script_4_Wards_4, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
+            tg.audioDialog(LocalizationKey.Script_4_Wards_5, LocalizationKey.Script_4_Wards_5, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
+            tg.immediate(_ => freezePlayerHero(true)),
+            tg.audioDialog(LocalizationKey.Script_4_Wards_6, LocalizationKey.Script_4_Wards_6, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
+
+            tg.fork([
+                tg.seq([
+                    tg.panCamera(playerHero.GetAbsOrigin(), cliffLocation1, _ => cameraSpeed),
+                    tg.wait(1),
+                    tg.panCamera(cliffLocation1, cliffLocation2, _ => cameraSpeed),
+                    tg.wait(1),
+                    tg.panCamera(cliffLocation2, cliffLocation3, _ => cameraSpeed),
+                    tg.wait(1),
+                    tg.panCamera(cliffLocation3, cliffLocation4, _ => cameraSpeed),
+                    tg.wait(1),
+                    tg.panCamera(cliffLocation4, playerHero.GetAbsOrigin(), _ => cameraSpeed),
+                ]),
+                tg.audioDialog(LocalizationKey.Script_4_Wards_7, LocalizationKey.Script_4_Wards_7, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
+            ]),
+            tg.immediate(_ => freezePlayerHero(false)),
             tg.immediate(_ => {
-                goalFetchWard.complete();
+                highlightUiElement(inventorySlot0UIPath);
```

Comment:
Chapter 2 Courier ends with a Daedalus, which usually would be placed in the first slot. This should be scalable for any slot that the wards are located in: find them using `GetItemInSlot()`.

Reply:
Added Daedalus in RequiredState then wards/dust will be in slot 1 now.